celery_progress/static/celery_progress/celery_progress.js

```diff
@@ -138,11 +139,23 @@ class CeleryProgressBar {
 
     async connect() {
         let response;
-        try {
-            response = await fetch(this.progressUrl);
-        } catch (networkError) {
-            this.onNetworkError(this.progressBarElement, this.progressBarMessageElement, "Network Error");
-            throw networkError;
+        let success = false;
+        let error = null;
+        let attempts = 0;
+        while(!success && attempts < this.maxNetworkRetryAttempts) {
+          try {
+              response = await fetch(this.progressUrl);
+              success = true;
+          } catch (networkError) {
+              error = networkError;
+              this.onNetworkError(this.progressBarElement, this.progressBarMessageElement, "Network Error");
+              attempts++;
+              await new Promise(r => setTimeout(r, 1000));
```

Comment:
sorry my async JS is not so good. How does this end up doing a retry? like how does it know what to call again?

Reply:
Ah right, duh. And this is essentially the equivalent of an async sleep?

Edit: Looks like yes https://stackoverflow.com/a/39914235/8207

Reply:
Oh yes, this was just not to spam the backend with requests every miliseconds or so.